src/nile/common.py

```diff
@@ -212,3 +227,9 @@ def get_hash(contract_name, overriding_path=None):
     return hex_class_hash(
         compute_class_hash(contract_class=contract_class, hash_func=pedersen_hash)
     )
+
+
+def call_cli(cmd_name, args, command_args):
+    """Make call to starknet_cli and return captured stdout."""
+    cmd = getattr(starknet_cli, cmd_name)
+    return capture_stdout(cmd(args=args, command_args=command_args))
```

Comment:
i think we can abstract all these CLI related functions (gateways, set_args, etc..) into a CLI class that also exposes `invoke` and `call` based on this one. `call_cli` was a proof of concept name but i don't really like it